src/template.rs

```diff
@@ -12,6 +13,9 @@ use serde::{Deserialize, Serialize};
 #[cfg(feature = "server")]
 const DEFAULT_TEMPLATE: &str = "main";
 
+#[derive(Serialize, Deserialize)]
+pub struct DynamicTemplate(pub String, pub String, pub String);
```

Comment:
This feels like it merits named fields rather than a tuple - hard to know what the strings mean and easy to get them mixed up!